server/app/controllers/admin/AdminProgramPreviewController.java

```diff
@@ -44,6 +60,23 @@ public Result preview(Request request, long programId) {
     }
   }
 
+  /**
+   * Creates and downloads a PDF for the given program. The PDF will contain all the blocks and the
+   * questions in each block.
+   */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result pdfPreview(Request request, long programId) throws ProgramNotFoundException {
+    ProgramDefinition program = programService.getFullProgramDefinition(programId);
+    ReadOnlyQuestionService roQuestionService =
+        questionService.getReadOnlyQuestionService().toCompletableFuture().join();
```

Comment:
(blocking) Have this method return `CompletionStage<Result>` and don't manually call to `join()`

Reply:
Yeah that looks right